src/contexts/__tests__/current-product.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { render, screen } from '@testing-library/react'
+import { renderHook } from '@testing-library/react-hooks'
+import { Product } from 'types/products'
+import {
+  CurrentProductProvider,
+  useCurrentProduct,
+} from 'contexts/current-product'
+
+/**
+ * Handles rendering both `CurrentProductProvider` and `useCurrentProduct` using
+ * the `renderHook` utility from `@testing-library/react-hooks`. Returns the
+ * result of `renderHook`.
+ */
+const setup = (currentProduct: Product) => {
+  const wrapper = ({ children }) => (
+    <CurrentProductProvider currentProduct={currentProduct}>
+      {children}
+    </CurrentProductProvider>
+  )
+  return renderHook(() => useCurrentProduct(), { wrapper })
+}
+
+describe('CurrentProductContext', () => {
```

Comment:
With the refactor away from effects, do we need a test case to validate that it sets the context value to `null` when the path is `/`?

Reply:
Good idea, I'll give that a try now 😊

Reply:
Added! What do you think? See https://github.com/hashicorp/dev-portal/pull/67/commits/96981029061087bb94433cd4587a727db843dff2

Reply:
Looks great, :shipit: !